spec/lib/appsignal/check_in/debouncer_spec.rb

```diff
@@ -0,0 +1,483 @@
+describe Appsignal::CheckIn::Debouncer do
+  include WaitForHelper
+  include TakeAtMostHelper
+
+  let(:transmitter) { Appsignal::Transmitter.new("http://checkin-endpoint.invalid") }
+
+  before do
+    allow(Appsignal).to receive(:active?).and_return(true)
+    allow(transmitter).to receive(:transmit).and_return(Net::HTTPSuccess.new("1.1", 200, "OK"))
+    allow(Appsignal::CheckIn).to receive(:transmitter).and_return(transmitter)
+    allow(Appsignal::CheckIn).to receive(:debouncer).and_return(subject)
+    # Shorten long debounce intervals to make the tests run faster.
+    stub_const("Appsignal::CheckIn::Debouncer::LONG_DEBOUNCE_SECONDS", 0.1)
+  end
+
+  after do
+    subject.stop
+  end
+
+  describe "when no event is sent" do
+    it "does not start a thread" do
+      expect(subject.thread).to be_nil
+    end
+
+    it "does not schedule a debounce" do
+      expect(subject.waker).to be_nil
+    end
+
+    it "can be stopped" do
+      # Set all debounce intervals to 10 seconds, to make the assertion
+      # fail if it waits for the debounce -- this ensures that what is being
+      # tested is that no debounces are awaited when stopping the debouncer.
+      stub_const("Appsignal::CheckIn::Debouncer::SHORT_DEBOUNCE_SECONDS", 10)
+      stub_const("Appsignal::CheckIn::Debouncer::LONG_DEBOUNCE_SECONDS", 10)
+
+      take_at_most(0.1) do
+        expect { subject.stop }.not_to raise_error
```

Comment:
This assertion doesn't do much. If it errors without this wrapped assertion, it will bubble up and stop the test with the error as well.

```suggestion
        subject.stop
```

Reply:
Yeah, `expect { ... }.not_to raise_error` never changes the outcome -- it's only there to indicate intent.

Reply:
After taking a second look at the tests that use this `expect { ... }.not_to raise_error` expectation, I'm leaning towards leaving it there, for the above reason -- otherwise, these errors have no expectations, and it's not very clear what their purpose is. Leave a comment if you disagree.